src/cfg/structured.rs

```diff
@@ -91,3 +105,131 @@ impl StructuredBlock {
         }
     }
 }
+
+impl StructuredFunction {
+    pub fn to_function(&self) -> Function {
+        let mut builder = StructuredCfgBuilder {
+            code: vec![],
+            scopes: vec![],
+            name_counter: 0,
+        };
+        self.block.to_code(&mut builder);
+
+        Function {
+            name: self.name.clone(),
+            args: self.args.clone(),
+            instrs: builder.code,
+            pos: None,
+            return_type: None,
+        }
+    }
+}
+
+pub struct StructuredCfgBuilder {
+    code: Vec<Code>,
```

Comment:
nit: can the name of this be plural to indicate that it's a vec? I found it confusing later when you do `builder.code.extend...` because the name `code` seems singular